### DISCUSSION TITLE
How to move existing connection origin/location

### DISCUSSION THREAD

**Discussion by tbyrk** [idea-statica/ideastatica-public#482] [Rhino/Grasshopper] [answered]
My goal is to export multiple connections to Tekla Structures from existing ideaCon files. I am able to do it but the origin is 0,0,0 for all of them. I have the connection locations in SAF so this could be automated.

<img width="582" alt="Origin point always 000" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/83bbbe9c-cba7-4600-ad67-ce01b582e94f">

I cannot seem to change it as there is no origin for the "Reference Connection" component. I added the input with Paint.

<img width="188" alt="Reference connection origin" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/1bd0ea86-962a-4751-9920-4ee6f99a2fa1">

Nor is there a "Location" input for "Modify connection point".

<img width="1279" alt="Move connection" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/da36654b-f6d9-48a9-a772-e36da6513186">

The end result should be that these three connections are at their correct positions.

<img width="991" alt="Connection point location" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/481551fb-fcbc-48ee-b2e1-f0d9613cfd84">

Any ideas?

Edit:

As a workaround of course I can just move the Tekla elements with vectors extracted from SAF but the connections will remain bunched up at 0,0,0 in Rhino. Not an elegant solution. Don't mind the messy connections. The script needs to adjusted for multiple connections.

<img width="1276" alt="Workaround" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/f97b4f85-8e0f-426f-be3a-742658fcebac">

**Comment by nathanlukeIDEA** [comment, marked as answer]
Hi @tbyrk,

It is a well-known issue and something we are currently looking to improve so that your goal can be achieved easily. 

Once a connection file is created from SAF/IOM the location of the original location is always reset to 0,0,0. Hence why when you retrieve the geometry again from the referenced connection it is reset to 0,0,0.  

Originally I did look to add the input as you described, however, currently, I would need to completely re-write the IOM with all translations and this should not be on the author of an external plugin to write considering the information was already previously defined. Therefore in the future, we will look to implement the current output of IDEA StatiCA connection geometry based on original creation geometry. 

With that said this nicer workflow will still likely be a little while away. I show below what I did in the recent webinar on the Grasshopper plugin: https://www.youtube.com/watch?v=4Rph3e3dG0Y&t=2065s.

If you turn off the preview of the get connection components it won't display the 0,0,0 geometry. Then merge (using trees the geometry you want to use for Tekla generation and translate by the vectors you already have. If you do it smartly you should still be able to map properties and geometry nicely to Tekla.
![image](https://github.com/idea-statica/ideastatica-public/assets/93578548/6fc713a2-b43b-4e94-8a26-656e214055a8)

**Comment by tbyrk** [comment]
One copy-paste from your script and a few wires and preview disables later I arrived at this. Thanks!

<img width="608" alt="Solution" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/5827ba5b-cf81-41c3-ba4d-a071344df365">


Looking forward to the nicer workflow as well.
